src/pool/srv_pool.c

```diff
@@ -6102,6 +6162,16 @@ pool_svc_update_map(struct pool_svc *svc, crt_opcode_t opc, bool exclude_rank,
 		D_GOTO(out, rc);
 	}
 
+	if (svc->ps_pool->sp_reint_mode == DAOS_REINT_MODE_NO_DATA_SYNC) {
+		D_DEBUG(DB_MD, "self healing is disabled for no_data_sync reintegration mode.\n");
+		rc = ds_pool_tgt_exclude_out(svc->ps_pool->sp_uuid, &target_list);
```

Comment:
ds_pool_tgt_exclude_out() is only for EXLUDE/DRAIN, but here pool_svc_update_map() possibly for other operations. if it is only for RINIT may need to check OPC but should use ds_pool_tgt_add_in() instead.
may confirm with @wangdi1 for what's the best way?

Reply:
Idea here is when targets are excluded, rebuild won't be triggered if mode is no_data_sync. but we want to mark failed targets state from down to downout, maybe need check opc is EXCLUDE/DRAIN.